federation-js/src/composition/validate/preComposition/noFed2Subgraphs.ts

```diff
@@ -0,0 +1,50 @@
+import { visit, GraphQLError, Kind } from 'graphql';
+import { ServiceDefinition } from '../../types';
+
+import { errorWithCode } from '../../utils';
+
+/**
+ * - There are no fields with @external on base type definitions
+ */
+export const noFed2Subgraphs = ({
+  name: serviceName,
+  typeDefs,
+}: ServiceDefinition) => {
+  const errors: GraphQLError[] = [];
+  visit(typeDefs, {
+    SchemaExtension(schemaExtensionNode) {
+      const directives = schemaExtensionNode.directives?.filter(directive => {
+        if (directive.name.value === 'link') {
```

Comment:
Nit: you can use `findDirectivesOnNode` for this, but it's really just equivalent / sugar for what you're doing.